pages/desktop/base.py

```diff
@@ -114,16 +120,22 @@ def click_login(self):
         from pages.desktop.login import Login
         return Login(self.selenium, self.page.base_url)
 
+    @property
+    def user_display_name(self):
+        self.wait.until(EC.visibility_of_element_located(self._user_locator))
+        return self.find_element(*self._user_locator)
+
     def click_logout(self):
-        user = self.find_element(*self._user_locator)
-        logout = self.find_element(*self._logout_locator)
         action = ActionChains(self.selenium)
+        user = WebDriverWait(self.selenium, 30, ignored_exceptions=StaleElementReferenceException) \
```

Comment:
I've been struggling a bit with this logout function because every now and then I would run into a `StaleElementReferenceException` for the `_user_locator` element. I've tried to countermeasure with a `ignored_exceptions` parameter inside the wait function to see how this goes.
@jrbenny35 maybe you can recommend a better solution?

Reply:
I'm using this to wait for the actual user `display_name` to be interactable before moving to hover over it, but I'm still not convinced that it helps much in this case. The login function that runs before the logout function already handles the page load aspect, so normally the element should already be interactable, but that appears to not be the case all the time.

Funny enough, I found out that you had an issue with the same function a few years back - https://bugzilla.mozilla.org/show_bug.cgi?id=1453779. Still, the solution that worked for you at that time is no longer working - i.e. `TypeError: rect is undefined` is reproducing again.

Reply:
Hmm yeah, that error is annoying. No clear way to fix it. I think if this works, then we just use it and put a comment with a link to that bug.

Reply:
I'm going to try a few more things locally to see if I can improve the stability in any way. Otherwise, I'm going to merge this as is and see how it goes.